lib/ensureNano.js

```diff
@@ -0,0 +1,52 @@
+function ensureNano(nano, db) {
+    var nanoDb = nano.use(db);
+    return wrapNano(nano, nanoDb, db);
+}
+
+function wrapNano(nano, nanoDb, db) {
+    var nanoObject = {};
+    Object.keys(nanoDb).forEach(function(method) {
+        if (typeof nanoDb[method] == "function") {
+            return nanoObject[method] = function() {
+                ensureDatabase.call(this, nano, nanoDb, db, method, arguments)
+            };
+        }
+
+        if (typeof nanoDb[method] == "object") {
+            return nanoObject[method] = wrapNano(nano, nanoDb[method], db);
+        }
+
+        return nanoObject[method] = nanoDb[method];
+    });
+
+    return nanoObject;
+}
+
+function ensureDatabase(nano, nanoDb, db, method, args) {
+    var originalCallback = replaceCallback(args, function(error) {
+        if (error && (error.reason == "Database does not exist." || error.reason == "no_db_file")) {
+            nano.db.create(db, function(error) {
+                if (error) {
+                    return originalCallback.apply(arguments);
+                }
+                replaceCallback(args, originalCallback);
+                nanoDb[method].apply(this, args)
+            })
+        }
+        else {
+            originalCallback.apply(this, arguments);
+        }
+    });
+    nanoDb[method].apply(this, args)
+}
+
+function replaceCallback(args, newCallback) {
+    var callback = function() {};
+    if (typeof args[args.length - 1] == "function") {
+        callback = args[args.length - 1];
+    }
+    args[args.length - 1] = newCallback;
```

Comment:
I think that the part I find tricky is that you modify the original `arguments` from https://github.com/cube-io/ensure-nano/pull/1/files#diff-8669b4c91a853bbba159d211d9ba6b8dR10 (line 10).